openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/OpenAPIRequestBodyMapper.java

```diff
@@ -242,31 +241,14 @@ private void handleArrayTypePayload(Map<String, Schema> schema, ArrayTypeDescrip
     private void handleMultipleMIMETypes(RequestBody bodyParameter,
                                          SeparatedNodeList<MappingFieldNode> fields,
                                          RequiredParameterNode payloadNode, Map<String, Schema> schema) {
+        List<String> mimeTypes = new ArrayList<>();
+        for (AnnotationNode annotation : payloadNode.annotations()) {
+            mimeTypes = extractAnnotationFieldDetails(HTTP_PAYLOAD, MEDIA_TYPE, annotation, semanticModel);
```

Comment:
Why are we reassigning values for `memeTypes` inside the for-loop in here? when we have multiple annotation, shouldn't we append the annotation data instead?

Reply:
Annotation in metadata is an array list. It can be service annotation, a user defines annotation so on. inside the for loop, we extract the annotation details and get the list of mimeTypes.  for avoiding repeating annotation check I moved this implement to its parent function 🙂